### PR TITLE
Add check and remediation for RHEL-07-040550 (shosts.equiv)

### LINKED ISSUE #1856: needs mapping/rule: SRG-OS-000480-GPOS-00227, SV-86903r1_rule, RHEL-07-040550 (no shosts.equiv)

### PR DESCRIPTION
**Pull request by ptitoliv**
#### Description:

OVAL check if there is any shosts.equiv files on the file system.
Remediation removes all the shosts.equiv files if there is a finding.

#### Rationale:

- Add check and remediation for shosts.equiv files detection.

- Fixes #1856

### CODE REVIEW

File: shared/checks/oval/no_host_based_files.xml [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+    </metadata>
+    <criteria operator="AND">
+      <criterion test_ref="test_no_shosts_equiv" negate="true" />
+    </criteria>
+  </definition>
+  <unix:file_test check="all" check_existence="at_least_one_exists" comment="look for shosts.equiv in /" id="test_no_shosts_equiv" version="1">
+    <unix:object object_ref="object_no_shosts_equiv_files_root" />
+  </unix:file_test>
+  <unix:file_object comment="look for any shosts.equiv file on the system" id="object_no_shosts_equiv_files_root" version="1">
+    <unix:behaviors recurse="directories" recurse_direction="down" max_depth="-1" recurse_file_system="all" />
```
Comment (redhatrises) [inline shared/checks/oval/no_host_based_files.xml]: Can you change `recurse_file_system="all"` to `recurse_file_system="local"`?

**Comment by ptitoliv** [comment]
@dahaic. I did a check on the whole system because this is what it was mentioned in the DISA STIG rule specified here. But I can update it if you want it only on /etc/shosts.equiv.

**Comment by dahaic** [comment]
Let's discuss it in #2524 - solution will be the same.

**Comment by dahaic** [comment]
@ptitoliv please apply same changes as you did in #2524 and that should be it :)

**Comment by ptitoliv** [comment]
@dahaic. I did some additional changes. There was a conflict with a recently added remediation script. And i also updated the check test from the XCCDF file.

**Comment by dahaic** [comment]
Thank you for the changes @ptitoliv, merging now :)

**Pull request merged**